### PR TITLE
Capture errors from runtime create failures

### LINKED ISSUE #492: no more useful output?
Try to create a container in a pod and use a non-existent command as the entrypoint (or command) like run `redis:alpine` (not `redis:latest`!!!) with `/bin/bash` (there's no `/bin/bash` in there) in there.
What you'll get is just:
```
$ sudo ./ocic ctr create --pod bd39 --name test_ctr10 --config test/testdata/container_config.json
Creating container failed: rpc error: code = 11 desc = EOF
```
We should be presenting something in the response as Kube (and Origin) will use this message in `kubectl describe pod <pod>` and that `EOF` is misleading, it should have been `/bin/bash` not found in `$PATH` or something like that.

(not sure you need https://github.com/kubernetes-incubator/cri-o/pull/491 to test this, you can reproduce with master easily afaict)

@mrunalp @cyphar PTAL

### PR DESCRIPTION
**Pull request by mrunalp**
Fixes #492 

@runcom @sameo @cyphar PTAL
Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

File: oci/oci.go [hunk 1]
```diff
@@ -130,14 +132,17 @@ func (r *Runtime) CreateContainer(c *Container, cgroupParent string) error {
...
+	}
 	cmd.ExtraFiles = append(cmd.ExtraFiles, childPipe)
 	// 0, 1 and 2 are stdin, stdout and stderr
 	cmd.Env = append(r.conmonEnv, fmt.Sprintf("_OCI_SYNCPIPE=%d", 3))
 
 	err = cmd.Start()
 	if err != nil {
 		childPipe.Close()
-		return err
+		return fmt.Errorf("failed to start command: %v")
```
Comment (sameo) [inline oci/oci.go]: `fmt.Errorf("failed to start command: %v", cmd)`

File: conmon/conmon.c [hunk 2]
```diff
@@ -475,8 +476,52 @@ int main(int argc, char *argv[])
...
 	}
-	if (!WIFEXITED(runtime_status) || WEXITSTATUS(runtime_status) != 0)
+	if (!WIFEXITED(runtime_status) || WEXITSTATUS(runtime_status) != 0) {
+		if (sync_pipe_fd > 0 && !exec) {
+			if (terminal) {
+				/* 
+				 * The stderr is captured in the parent when terminal is passed down.
+				 * Just send pid as -1 to indicate create container issue to parent.
+				 */
+				len = snprintf(buf, BUF_SIZE, "{\"pid\": %d}\n", -1);
```
Comment (sameo) [inline conmon/conmon.c]: Why the `-1` here ?

Comment (sameo) [inline conmon/conmon.c]: nm, it's described in the above comment...

`Just send pid as -1 to indicate create container issue to parent.`

Comment (sameo) [inline conmon/conmon.c]: So could we keep the pid information and use `message != ""` to signal an error ?

Comment (mrunalp) [inline conmon/conmon.c]: I have moved the comments to hopefully make them better.

File: oci/oci.go [hunk 3]
```diff
@@ -171,15 +176,45 @@ func (r *Runtime) CreateContainer(c *Container, cgroupParent string) error {
...
+			}
+		}
 	case <-time.After(ContainerCreateTimeout):
 		return fmt.Errorf("create container timeout")
 	}
 	return nil
 }
 
+// Remove conmon debug messages from passed string
+func sanitizeErrorMessage(errString string) string {
```
Comment (sameo) [inline oci/oci.go]: I'd call that one `sanitizeConmonErrorMessage()` to make it less generic.

**Comment by sameo** [comment]
Besides the sanitize function naming, LGTM.

**Comment by sameo** [comment]
And FWIW this PR already came handy with debugging CRI-O + [Clear Containers 3.0](https://github.com/clearcontainers/runtime) :)

**Comment by runcom** [comment]
I'd run kube tests on this, LGTM overall

**Comment by runcom** [comment]
@mrunalp could you add an simple integration test for a command not found error?

**Comment by runcom** [comment]
(running kube e2e-node conformance against latest kube master now, will report back, and, not blocking, I'll run tests for kube-1.6.1 as well later)

**Comment by mrunalp** [comment]
@runcom Cool, I'll add the integration test and address @sameo comment. k8s tests are really important for this one :+1:

**Comment by runcom** [comment]
@mrunalp actually, I'm going to test with kube-1.6.x since kubernetes upstream master upgraded to go1.8.1 and many deps changed also and using `dep` is not making the update easy :(

**Comment by mrunalp** [comment]
Okay sounds good.

File: oci/oci.go (hunk 3 above)
Comment (TomSweeneyRedHat) [inline oci/oci.go]: 👍

**Comment by TomSweeneyRedHat** [comment]
LGTM, assuming @runcom is happy with the testing

**Comment by mrunalp** [comment]
This passes the k8s e2e tests rebased on top of master.

**Comment by runcom** [comment]
cool, I'll merge after tests.

**Pull request merged**